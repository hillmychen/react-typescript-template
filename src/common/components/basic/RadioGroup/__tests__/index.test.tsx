import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { RadioGroup } from '../index'

describe('RadioGroup', () => {
  test('test RadioGroup', () => {
    const { asFragment, getByText } = render(
      <RadioGroup
        value={1}
        dataSource={[
          { label: '男', value: 1 },
          { label: '女', value: 2 },
          { label: '保密', value: 3, disabled: true },
        ]}
      />
    )
    expect(getByText('男').closest('label')!.querySelector('input')).toBeChecked()
    expect(getByText('保密').closest('label')!.querySelector('input')).toBeDisabled()
    expect(asFragment()).toMatchSnapshot()
  })
})
