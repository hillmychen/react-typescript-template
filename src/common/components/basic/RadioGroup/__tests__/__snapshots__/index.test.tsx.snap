// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup test RadioGroup 1`] = `
<DocumentFragment>
  <div
    class="ant-radio-group ant-radio-group-outline radioBox"
  >
    <label
      class="ant-radio-wrapper ant-radio-wrapper-checked"
    >
      <span
        class="ant-radio ant-radio-checked"
      >
        <input
          checked=""
          class="ant-radio-input"
          type="radio"
          value="1"
        />
        <span
          class="ant-radio-inner"
        />
      </span>
      <span>
        男
      </span>
    </label>
    <label
      class="ant-radio-wrapper"
    >
      <span
        class="ant-radio"
      >
        <input
          class="ant-radio-input"
          type="radio"
          value="2"
        />
        <span
          class="ant-radio-inner"
        />
      </span>
      <span>
        女
      </span>
    </label>
    <label
      class="ant-radio-wrapper ant-radio-wrapper-disabled"
    >
      <span
        class="ant-radio ant-radio-disabled"
      >
        <input
          class="ant-radio-input"
          disabled=""
          type="radio"
          value="3"
        />
        <span
          class="ant-radio-inner"
        />
      </span>
      <span>
        保密
      </span>
    </label>
  </div>
</DocumentFragment>
`;
